pydantic/main.py

```diff
@@ -186,7 +186,7 @@ def __new__(mcs, name, bases, namespace):
             for ann_name, ann_type in annotations.items():
                 if is_classvar(ann_type):
                     class_vars.add(ann_name)
-                elif not ann_name.startswith('_') and ann_name not in namespace:
+                elif (not ann_name.startswith('_') or '__root__' == ann_name) and ann_name not in namespace:
```

Comment:
we do this at least three times, can we have a function for it.

Reply:
I have created ``is_valid_field function`` for checking a field name.